### ISSUE TITLE
Durable entity state changes in a finally block not saved when an unhandled exception occurs

### ISSUE THREAD

**Issue by SecretPocketCat** [Azure/azure-functions-durable-extension#1684]
### Description
Durable entity state changes in a finally block not saved when an unhandled exception occurs.  Could repro that both on Azure and locally.

### Expected behavior

- An unhandled exception occurs in a durable entity operation
- The finally block changes the state of the entity
- The state is saved

### Actual behavior

- The state changes are saved

### Relevant source code snippets
```csharp
        public async Task SendMessagesAsync()
        {
            try
            {
                await _chatService.SendMessagesAsync(Messages);
            }
            finally
            {
                Messages.Clear();
            }
        }
```

### Known workarounds
Handle the exceptions

### App Details

- **Durable Functions extension version**: 2.4.0
- **Azure Functions runtime version**: 2.0
- **Programming language used**: C#

### If deployed to Azure

- **Orchestration instance ID(s)**: @usertomerchantmessagesentity@27be3ce0-bf21-4879-41c0-08d82bd9c191

**Comment by cgillum** [comment]
I believe it may be by design that all state changes are discarded when an entity operation fails with an unhandled exception but I'll need @sebastianburckhardt to confirm. I could not find the expected behavior mentioned in the documentation.

**Comment by mpaul31** [comment]
Is this exposed via host.json or manual configuration?

RollbackEntityOperationsOnExceptions

**Comment by sebastianburckhardt** [comment]
Yes, `RollbackEntityOperationsOnExceptions` is the setting you want to change, it defaults to true.
You can set this in host.json I believe.

**Comment by sebastianburckhardt** [comment]
For more context on why we added the "rollback on exceptions" feature, see the discussion in #1223 and #1219.

Basically, it makes entity operations behave like transactions, which is often (but maybe not always) desirable.

It does appear that we have not documented this feature though, so that needs to be done.

**Comment by SecretPocketCat** [comment]
@sebastianburckhardt Great, that makes a lot of sense, thank you. I was just missing a mention in the docs.
